Dart.gitignore

```diff
@@ -3,11 +3,27 @@
 # Files and directories created by pub
 .dart_tool/
 .packages
-.pub/
 build/
 # If you're building an application, you may want to check-in your pubspec.lock
 pubspec.lock
 
 # Directory created by dartdoc
 # If you don't generate documentation locally you can remove this line.
 doc/api/
+
+# Avoid committing generated Javascript files:
+*.dart.js
+*.info.json      # Produced by the --dump-info flag.
+*.js             # When generated by dart2js. Don't specify *.js if your
+                 # project includes source files written in JavaScript.
+*.js_
+*.js.deps
+*.js.map
+
+
+# files and directories created by development environments
```

Comment:
These are not specific to Dart, and are maintained in `Global/JetBrains.gitignore` or `Global/macOS.gitignore`.

Please tidy this section up.